wss/wss.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wss
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	info "github.com/google/cadvisor/info/v1"
+	"github.com/google/cadvisor/stats"
+	"k8s.io/klog"
+)
+
+const (
+	cgroupProcfs = "cgroup.procs"
+)
+
+var (
+	smapsFilePathPattern     = "/proc/%s/smaps"
+	clearRefsFilePathPattern = "/proc/%s/clear_refs"
+
+	referencedRegexp = regexp.MustCompile(`Referenced:\s*([0-9]+) kB`)
+	isDigitRegExp    = regexp.MustCompile("\\d+")
+)
+
+type wssManager struct {
+	resetInterval uint64
+	wssEnabled    bool
```

Comment:
I think that you do not need this property - if it's `false` then you can return `stats.NoopManager` that will always return `stats.NoopColector` :)